src/wamr/wamr.cc

```diff
@@ -261,9 +261,11 @@ bool Wamr::link(std::string_view /*debug_name*/) {
     const wasm_name_t *name_ptr = wasm_importtype_name(import_types.get()->data[i]);
     const wasm_externtype_t *extern_type = wasm_importtype_type(import_types.get()->data[i]);
 
-    std::string_view module_name(module_name_ptr->data, module_name_ptr->size);
-    std::string_view name(name_ptr->data, name_ptr->size);
-    assert(name_ptr->size > 0);
+    assert(std::strlen(name_ptr->data) > 0);
+
+    std::string_view module_name(module_name_ptr->data, std::strlen(module_name_ptr->data));
+    std::string_view name(name_ptr->data, std::strlen(name_ptr->data));
```

Comment:
I'm confused by those changes.

Did `wasm_importtype_name` / `wasm_importtype_name` / `wasm_exporttype_name` stop setting the correct `size` in the returned `wasm_name_t`? If so, that sounds like a breaking change with Wasm C/C++ API.

Also, I don't believe those are guaranteed to be NULL-terminated strings.

Reply:
In WAMR, we provid c-style strings now

Reply:
Oh, so the `wasm_name_t`'s `size` now includes the `NULL` byte and cannot be used here? That seems like a bad design.

Also, if WAMR made this change, but other Wasm C/C++ API libraries did not, then this breaks the API compatibility in a hard to debug way. This doesn't matter for us, since we do per-engine integration, but technically that's against the core idea of Wasm C/C++ API where you should be able to compile the code once and then load any Wasm C/C++ API compatible library at runtime.

But I guess the ship has sailed on this, so we need to make the adjustments here.

Code-wise, I think you should remove the second parameter, and let `std::string_view`'s constructor calculate the size of `NULL`-terminated string itself.